packages/migrate/test/migrate.test.ts

```diff
@@ -0,0 +1,430 @@
+import { readFileSync } from 'node:fs';
+import path, { dirname, resolve } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import { Project } from 'fixturify-project';
+import { Reporter } from '@rehearsal/reporter';
+import { afterEach, beforeEach, describe, expect, test } from 'vitest';
+import { Logger, createLogger, format, transports } from 'winston';
+import { MigrateInput, migrate } from '../src/migrate.js';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+const projectPath = resolve(__dirname, 'fixtures', 'project');
+
+function changeExtension(file: string, ext: string): string {
+  const parts = path.parse(file);
+
+  ext = ext.startsWith('.') ? ext : `.${ext}`;
+
+  return path.format({
+    ...parts,
+    ext,
+    base: undefined,
+  });
+}
+
+const extLookup = {
+  '.js': '.ts',
+  '.gjs': '.gts',
+  '.hbs': '.hbs',
+  '.ts': '.ts',
+};
+
+type ValidExtension = keyof typeof extLookup;
+
+const validExtensions = Object.keys(extLookup) as Array<ValidExtension>;
+
+function isValidExtension(ext: string): ext is ValidExtension {
+  return validExtensions.includes(ext);
+}
+
+function expectFile(filePath: string): Vi.Assertion<string> {
+  return expect(readFileSync(filePath, 'utf-8'));
+}
+
+function prepareInputFiles(project: Project, files: string[] = ['index.js']): string[][] {
+  const inputs = files.map((file) => {
+    return resolve(project.baseDir, 'src', file);
+  });
+
+  const outputs = inputs.map((file) => {
+    const fileExt = path.extname(file);
+
+    if (!isValidExtension(fileExt)) {
+      throw new Error(`found a file extension we don't know about: ${fileExt}`);
+    }
+
+    const ext = extLookup[fileExt];
+
+    return changeExtension(file, ext);
+  });
+
+  return [inputs, outputs];
+}
+
+describe('migrate', () => {
+  describe('.gjs', () => {
```

Comment:
Can we add some cases where we have a `class` based implementation?

Reply:
There's a `with-class` case in this section that does have a class-based component with an inlined `<template>`. Is that what you mean?